core/src/test/kotlin/com/amplitude/core/utilities/HttpClientTest.kt

```diff
@@ -0,0 +1,108 @@
+package com.amplitude.core.utilities
+
+import com.amplitude.core.Configuration
+import com.amplitude.core.events.BaseEvent
+import io.mockk.every
+import io.mockk.spyk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.json.JSONObject
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import java.util.concurrent.TimeUnit
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class HttpClientTest {
+    private lateinit var server: MockWebServer
+    val apiKey = "API_KEY"
+    val clientUploadTimeMillis = 1699905773000L
+    val clientUploadTimeString = "2023-11-13T20:02:53.000Z"
+
+    @ExperimentalCoroutinesApi
+    @BeforeEach
+    fun setup() {
+        server = MockWebServer()
+        server.start()
+    }
+
+    @AfterEach
+    fun shutdown() {
+        server.shutdown()
+    }
+
+    @Test
+    fun `test client_upload_time is set on the request`() {
+        server.enqueue(MockResponse().setBody("{\"code\": \"success\"}"))
+
+        val config = Configuration(
+            apiKey = apiKey,
+            serverUrl = server.url("/").toString()
+        )
+        val event = BaseEvent()
+        event.eventType = "test"
+
+        val httpClient = spyk(HttpClient(config))
+        every { httpClient.getCurrentTimeMillis() } returns clientUploadTimeMillis
+
+        val connection = httpClient.upload()
+        System.currentTimeMillis()
```

Comment:
It seems that this is not used? And the same for the other test.

Reply:
Good catch. Thanks